Add IQS031 rule (#2396)
home-assistant/developers.home-assistant: docs/core/integration-quality-scale/rules/iqs031.md

---
title: "IQS031 - Icon translations"
---

IQS031 is as follows:

Icon translations

## Reasoning

In the past, icons were part of the state of the integration.
This was not really necessary, as they were usually either static or had a fixed set of states.

> To relief the state machine, icon translations were introduced.

Reviewer: ```suggestion
To relieve the state machine, icon translations were introduced.
```

The name of this feature sounds weird since it is not about translating the icon itself, but rather referencing an icon by a translation key.
The idea behind icon translations is that the integration defines icons in a file, which is then used by the frontend to display the icon.

> This also adds support for icons for the possible preset modes of a climate entity to pick from for example.

Reviewer: ```suggestion
This also adds support for different icons for state attribute values, for example the possible preset modes of a climate entity.
```

> Since adding icon translations to an integration betters the consistency between integrations, we consider it a best practice to use them.
> 

Reviewer: I don't think it has anything to do with consistency, besides following the IQS rules will improve consistency in general.
```suggestion
```

:::info
Be aware that entities can also get icons from the device class.
If the context of the entity is exactly the same as the device class, we should not overwrite this icon to maintain consistency between integrations.

> So for example, a PM2.5 sensor entity would not get a custom icon, as the device class already provides it in the same context.

Reviewer: ```suggestion
For example, a PM2.5 sensor entity would not get a custom icon, as the device class already provides it in the same context.
```
